packages/signature-controller/src/SignatureController.ts

```diff
@@ -880,4 +897,67 @@ export class SignatureController extends BaseController<
 
     return networkClient.configuration.chainId;
   }
+
+  #decodePermitSignatureRequest(
+    signatureRequestId: string,
+    request: OriginalRequest,
+    chainId: string,
+  ) {
+    // Code below will invoke signature request decoding api for permits
+    if (!this.#decodingApi) {
+      return;
+    }
+    let decodedRequest: DecodedRequestInfo;
+    try {
+      const { primaryType } = JSON.parse(request.params?.[1] ?? '');
+      if (primaryType === 'Permit') {
+        this.#updateMetadata(signatureRequestId, (draftMetadata) => {
+          draftMetadata.decodedRequest = 'IN_PROGRESS';
+        });
+        const { method, origin, params } = request;
+        fetch(`${this.#decodingApi}?chainId=${chainId}`, {
+          method: 'POST',
+          body: JSON.stringify({
+            method,
+            origin,
+            params: [
+              params?.[0],
+              JSON.parse(
+                convertNumbericValuestoQuotedString(params?.[1]) ?? '',
+              ),
+            ],
+          }),
+          headers: { 'Content-Type': 'application/json' },
+        })
+          .then((response) => {
+            return response.json();
+          })
+          .then((result) => {
```

Comment:
Rather than relying on `then` and `catch`, would it be easier for this method to be `async`?

Reply:
I preferred then/catch here to help us keep this method async while processing signature request

Reply:
I actually do not want the flow to wait for decoding api to return.

Reply:
I understand that, but this method can still be `async` without blocking the flow, we just wouldn't `await` when we call it.